view/frontend/templates/js/replacejs.phtml

```diff
@@ -753,85 +755,56 @@ ob_start();
             } else {
-            // define the params sent with the request variable
-            var params = [];
-            params.push('form_key=' + $('[name="form_key"]').val());
-            params = params.join('&');
-
             var hintBarrier = boltBarrier();
+            var cartBarrier = boltBarrier();
 
-            cart = new Promise(function (resolve, reject) {
-
-                createRequest = true;
-
-                // send create order request
-                $.get(settings.create_order_url, params)
-                    .done(function (data) {
+            customerData.get('boltcart').subscribe(function(data) {
+                cart = data.cart !== undefined ? data.cart : {};
+                if (JSON.stringify(cart) === oldBoltCartValue) {
+                    return;
+                }
 
                         var boltRestricted = !!data.restrict;
 
                         toggleCheckoutIfNeeded(boltRestricted);
 
                         // Stop if Bolt checkout is restricted
                         if (boltRestricted) {
-                            if (popUpOpen) reject(new Error(data.message));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
-
-                        if (data.status !== 'success') {
-                            reject(new Error(data.message || 'Network request failed'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
-
-                        if (!data.cart) {
-                            reject(new Error('The cart info is missing.'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
-
-                        if (!data.cart.orderToken) {
-                            reject(new Error('The cart is empty.'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
+                            cart = {};
+                            hints = {};
+                        } else {
 
                         var prefill = isObject(data.hints.prefill) ? deepMergeObjects(hints.prefill, data.hints.prefill) : hints.prefill;
 
                         hints = deepMergeObjects(hints, data.hints);
                         hints.prefill = prefill;
                         //////////////////////////
-
-                        resolve(data.cart);
-                        hintBarrier.resolve(hints);
+                        }
+                        if (oldBoltCartValue === "") {
+                            // resolve promises for initial .configure() call
+                            cartBarrier.resolve(cart);
+                            hintBarrier.resolve(hints);
+                        } else {
+                            // reconfigure bolt checkout with new values
+                            config = {};
```

Comment:
maybe put a short comment explaining why it's empty?

Reply:
All functionality works after this PR except the always shown button. I handle it in the next PR and remove this assignment. Sorry for this confusing things.

Reply:
was this resolved?

Reply:
Yes, I added comment